cmd/populator/cmd/register/ocf_manifests.go

```diff
@@ -89,30 +87,83 @@ func runDBPopulate(ctx context.Context, src string) (err error) {
 			err = multierror.Append(err, cErr)
 		}
 	}()
-
 	session := driver.NewSession(neo4j.SessionConfig{})
 	defer func() {
 		if sErr := session.Close(); sErr != nil {
 			err = multierror.Append(err, sErr)
 		}
 	}()
 
-	var gitHash []byte
-	if cfg.UpdateOnGitCommit {
-		log.Info("APP_UPDATE_ON_GIT_COMMIT set. Updating manifests only if git commit changed.")
-		gitHash, err = getGitHash(rootDir)
+	dataInDB, err := dbpopulator.IsDataInDB(session, log, commits)
+	if err != nil {
+		return errors.Wrap(err, "while verifying commits in db")
+	}
+	if dataInDB {
+		return nil
+	}
+	for _, src := range sourcesInfo {
+		err = runDBPopulate(ctx, cfg, session, log, src)
+		if rErr := os.RemoveAll(src.dir); rErr != nil {
+			err = multierror.Append(err, rErr)
+		}
+		if err != nil {
+			return errors.Wrap(err, "while populating db")
+		}
+	}
+	err = dbpopulator.SaveCommitsMetadata(session, commits)
+	if err != nil {
+		return errors.Wrap(err, "while saving metadata into db")
+	}
+
+	return nil
+}
+
+func getSourcesInfo(ctx context.Context, cfg dbpopulator.Config, log *zap.Logger, sources []string) ([]sourceInfo, error) {
+	var sourcesInfo []sourceInfo
+	for _, source := range sources {
+		parent, err := ioutil.TempDir("/tmp", "*-hub-parent")
```

Comment:
it will be good to have a single directory for all downloaded manifests, it will also make the clean-up easier :)

```
/tmp/3595623673-hub-parent/hub/manifests/*
```

other folder
```
/tmp/920551000-hub-parent/hub/manifests/*
```

can become:
```
/tmp/920551000-sources-parent/{encoded_source_name}/manifests/
```

Reply:
It is actually a little bit tricky because the `getter.Download` needs to have a catalog that does not exist yet. (that is one of the reasons of existence `hub` in the path). 

Anyhow, I correct it in the way that we have:
```
/tmp/2911003352-hubs-parent/{encoded_source_name}/manifests/
```

when someone would like to run the populator locally with the `.` (edge case), then we instead of using encoded source name use random dir name:
```
/tmp/2911003352-hubs-parent/edd107008e5ad93f757e222c335c8e19-local/manifests/
```